### ISSUE TITLE
Update Rust

### ISSUE THREAD

**Issue by lidavidm** [HaliteChallenge/Halite-III#175]

**Comment by lidavidm** [comment]
Generally Rust will statically link other Rust libraries, including the standard library, since there's not yet a defined ABI. And Rust generally tries to maintain backwards compatibility, though IIRC minor things have broken in the past. They are introducing breaking changes in the latest compiler, but authors have to opt-in to them at the project level.

I could go through the current Rust bots and make sure they still compile and run after the update (there seem to be about 50 currently)

**Comment by Janzert** [comment]
I don't think you need to bother checking that they will all still compile without any change. From what you said it sounds like it should be good and if something does break there it will be immediately obvious when the person resubmits.

It also sounds pretty safe that the already compiled bots will continue running without a problem after the upgrade. I would be more inclined to try running those current binaries on an upgraded server or at least keeping a close eye on them right after the upgrade.
